sdk/storage/azure-storage-blob-changefeed/azure/storage/blob/changefeed/_models.py

```diff
@@ -119,36 +121,58 @@ def __next__(self):
+        self._segment_paths_generator = self._get_segment_paths(start_year=start_year)
+        next_segment_path = next(self._segment_paths_generator)
 
-        # if start_time is specified, remove all segments earlier than start_time
+        # if start_time is specified, skip all segments earlier than start_time
         if self.start_time:
-            while self.unprocessed_segment_paths and \
-                    self._is_earlier_than_start_time(self.unprocessed_segment_paths[0]):
-                self.unprocessed_segment_paths.popleft()
+            while next_segment_path and self._is_earlier_than_start_time(next_segment_path):
+                next_segment_path = next(self._segment_paths_generator)
 
         # if change_feed_cursor is specified, start from the specified segment
         if cf_cursor:
-            while self.unprocessed_segment_paths[0].name != cf_cursor['segment_path']:
-                self.unprocessed_segment_paths.popleft()
+            while next_segment_path and next_segment_path != cf_cursor['segment_path']:
+                next_segment_path = next(self._segment_paths_generator)
 
-        self.current_segment = self._get_next_segment(self.page_size,
-                                                      segment_cursor=cf_cursor['segment_cursor'] if cf_cursor else None)
+        self.current_segment = self._get_next_segment(
+            next_segment_path,
+            self.page_size,
+            segment_cursor=cf_cursor['segment_cursor'] if cf_cursor else None)
 
-    def _get_next_segment(self, page_size, segment_cursor=None):
-        if self.unprocessed_segment_paths:
-            segment_path = self.unprocessed_segment_paths.popleft()
+    def _get_next_segment(self, segment_path, page_size, segment_cursor=None):
+        if segment_path:
             if self.end_time and self._is_later_than_end_time(segment_path):
-                self.unprocessed_segment_paths = []
                 return None
             return Segment(self.client, segment_path, page_size, segment_cursor)
         return None
 
+    def _get_segment_paths(self, start_year=""):
+        cur_year = datetime.today().year
+        while not start_year or start_year <= cur_year:
+            paths = self.client.list_blobs(name_starts_with=SEGMENT_COMMON_PATH + str(start_year))
+            for path in paths:
+                yield path.name
+
+            # if not searching by prefix, all paths would have been iterated already, so it's time to yield None
+            if not start_year:
+                break
+            # search the segment prefixed with next year.
+            start_year += 1
+        yield None
+
     @staticmethod
     def _parse_datetime_from_segment_path(segment_path):
-        path_tokens = segment_path.name.split("/")
+        path_tokens = segment_path.split("/")
         return datetime(int(path_tokens[2]), int(path_tokens[3]), int(path_tokens[4]), int(path_tokens[5][:2]))
```

Comment:
I'd introduce some local variables to make this code more self descriptive.

Reply:
will do!